File: src/types/application.ts
export interface Application {
  id: string;
  bundle_id?: string;
  name: string;
  display_name: string;
  version?: string;
  developer?: string;
  bundle_path: string;
  executable_path?: string;
  icon_path?: string;
  is_system: boolean;
  is_running: boolean;
  size_bytes: number;
}

export type ArtifactConfidence = "Unknown" | "Low" | "Medium" | "High" | "Exact";

export interface Artifact {
  path: string;
  category: string;
  confidence: ArtifactConfidence;
  size_bytes: number;
}

File: src-tauri/src/models/application.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Application {
    pub id: String,
    pub bundle_id: Option<String>,
    pub name: String,
    pub display_name: String,
    pub version: Option<String>,
    pub developer: Option<String>,
    pub bundle_path: String,
    pub executable_path: Option<String>,
    pub icon_path: Option<String>,
    pub is_system: bool,
    pub is_running: bool,
    pub size_bytes: u64,
}
